packages/core/src/interfaces/objects/QueryMetadata.ts

```diff
@@ -0,0 +1,17 @@
+import {
+  EWalletDataType,
+  IpfsCID,
+  JSONString,
+} from "@snickerdoodlelabs/objects";
+
+export class QueryMetadata {
+  public constructor(
+    public name: string,
```

Comment:
If you're going to make an object for the query metadata separate from the complete query, you should at least include the ipfs CID as a field so that you know what query this metadata pertains to.